aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -151,9 +149,17 @@ static inline void launch_vectorized_kernel(
   constexpr auto io_size = calc_io_size<func_t>();
   int64_t grid = (N + io_block_work_size<io_size>() - 1) / io_block_work_size<io_size>();
   auto stream = at::cuda::getCurrentCUDAStream();
-  int vec_size = memory::can_vectorize_up_to<func_t>(data);
+  using cpp_type = typename function_traits<func_t>::result_type;
+  const uint16_t max_vec_size = memory::can_vectorize_up_to<func_t>(data, sizeof(cpp_type) == 2);
```

Comment:
maybe leave a comment explaining that this is setting the `use_vec8` bool and how or why 1-byte dtypes should not be vectorized to 8